add density scaling performance test, ranging from 200, 400, and 600 VMis

marceloamaral opened the pull request:
Add a Prow job that runs a medium-scale density scaling performance test once a day by creating 200, 400, up to 600 VMIs in the performance cluster.

The results of this work will be used as the baseline performance of KubeVirt to evaluate a possible performance regression.

Also, I updated the density test to use golang images as the kubevirt build was failing because the bootstrap images don't have go.

This PR depends on kubevirt/kubevirt/pull/7060

/cc @fgimenez @rmohr @davidvossel @rthallisey

Signed-off-by: Marcelo Amaral <marcelo.amaral1@ibm.com>

fgimenez commented on github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:
```diff
@@ -1123,6 +1123,105 @@ periodics:
+    preset-dind-enabled: "true"
+    preset-docker-mirror-proxy: "true"
+    preset-kubevirtci-quay-credential: "true"
+    preset-shared-images: "true"
```
This preset is only required for windows and rhel tests

fgimenez commented on github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:
```diff
@@ -1123,6 +1123,105 @@ periodics:
+        value: Always
+      - name: GIT_ASKPASS
+        value: ../project-infra/hack/git-askpass.sh
+      image: quay.io/kubevirtci/bootstrap:v20210906-994b913
```
This image tag is pretty old, could you please add a newer one? From https://quay.io/repository/kubevirtci/bootstrap?tab=tags, `v20220110-4643253` should be ok

marceloamaral replied:
Actually, building the kubevirt code was failing since bootstrap image does not have go installed.
I changed to golang image...

fgimenez commented on github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:
```diff
@@ -1123,6 +1123,105 @@ periodics:
+      - /bin/sh
+      - -c
+      - |
+        # install yq
```
This is a lot of code, given that you need to propose `kubevirt-density-scale.yaml` to kubevirt/kubevirt, could you please also add this script under `hack`?

marceloamaral replied:
Done!

marceloamaral commented:
@fgimenez I changed the PR as you requested.

Also, I updated the density test to use golang images as the kubevirt build was failing because the bootstrap images don't have go.

dhiller commented:
@marceloamaral this is the first test of the job on the cluster: https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/logs/periodic-kubevirt-performance-cluster-scale-density-test/1484491434689040384

dhiller commented on github/ci/prow-deploy/files/jobs/kubevirt/kubevirt/kubevirt-periodics.yaml:
```diff
@@ -1059,39 +1055,102 @@ periodics:
+  labels:
+    preset-bazel-cache: "true"
+    preset-bazel-unnested: "true"
+    preset-dind-enabled: "true"
```
```suggestion
    preset-dind-enabled: "true"
    preset-kubevirtci-quay-credential: "true"
```

dhiller replied:
You would need to the quay credentials preset to have the env vars available in the job.

marceloamaral replied:
Thanks @dhiller!
I updated it, could you please review it again?

dhiller reviewed (commented):
/approve
/hold

Job lgtm, remove hold when everyone is ok with this.

marceloamaral commented:
/unhold
